appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java

```diff
@@ -2186,27 +2132,28 @@ protected void postInvoke(EjbInvocation inv, boolean doTxProcessing) {
             releaseContext(inv);
         }
 
-        if ( inv.exception != null ) {
+        if (inv.exception != null) {
 
             // Unwrap the PreInvokeException if necessary
-            if ( inv.exception instanceof PreInvokeException ) {
-                inv.exception = ((PreInvokeException)inv.exception).exception;
```

Comment:
I changed the tiny PreInvokeException wrapper to use standard mechanism - it already is wrapper.